cranelift/codegen/src/isa/aarch64/inst.isle

```diff
@@ -1543,3 +1561,41 @@
 
 ;; 64-bit passthrough.
 (rule (put_in_reg_zext64 val @ (value_type $I64)) (put_in_reg val))
+
+;; Misc instruction helpers ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(decl trap_if_zero (Reg) Reg)
+(rule (trap_if_zero reg)
+      (let ((_ Unit (emit (MInst.TrapIf (cond_br_zero reg) (trap_code_division_by_zero)))))
```

Comment:
It is a little strange that `trap_if_zero` unconditionally uses the `trap_code_division_by_zero` trap code. `trap_if_zero`'s name doesn't suggest anything about division. Could we either workshop the name into something like `trap_if_zero_divisor` or make it take a trap code parameter?